### PR TITLE
[JIT] End-to-end example-based robustness testing for hybrid frontend

### PR DESCRIPTION
**Pull request by jamesr66a**
Put these in `test_jit` for two reason:

1) They'll get run under CI
2) We have existing machinery here to test traced models and their gradients

These tests revealed many issues, which I've created issues for and will shortly create a meta-issue to track.

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -3026,6 +3043,404 @@ def fn(x):
...
+                return x
+
+            def _initialize_weights(self):
+                init.orthogonal_(self.conv1.weight, init.calculate_gain('relu'))
+                init.orthogonal_(self.conv2.weight, init.calculate_gain('relu'))
+                init.orthogonal_(self.conv3.weight, init.calculate_gain('relu'))
+                init.orthogonal_(self.conv4.weight)
+
+        net = Net(upscale_factor=4)
+        net._initialize_weights()
```
Comment (apaszke) [inline test/test_jit.py]: This looks unnecessary. Can you remove `_initialize_weights` altogether? It doesn't even take part in tracing

File: test/test_jit.py [hunk 2]
```diff
@@ -3026,6 +3043,404 @@ def fn(x):
...
+
+    def test_reinforcement_learning(self):
+        class Policy(nn.Module):
+            def __init__(self):
+                super(Policy, self).__init__()
+                self.affine1 = nn.Linear(4, 128)
+                self.affine2 = nn.Linear(128, 2)
+
+                self.saved_log_probs = []
+                self.rewards = []
```
Comment (apaszke) [inline test/test_jit.py]: Those are unnecessary

**Review by apaszke — APPROVED** [review]
LGTM. Some code is unnecessary and could be removed.

**Comment by jamesr66a** [comment]
@zdevito  i think that one fusion test was indeed flaky; i don't see the failure anymore. I'm going to go ahead and merge
